jobs/build-charms/charms.py

```diff
@@ -337,30 +346,40 @@ def promote(self, charm_entity, from_channel, to_channels):
         self._echo(
             f"Promoting :: {charm_entity:^35} :: from:{from_channel} to: {to_channels}"
         )
-        if from_channel == "unpublished":
+        if "unpublished" == from_channel:
             charm_status = self._unpublished_revisions(charm_entity)
         else:
             charm_status = [
                 row
                 for row in self.status(charm_entity)
-                if f"{row['Track']}/{row['Channel']}" == from_channel
+                if all(
+                    (
+                        row["Revision"],
+                        f"{row['Track']}/{row['Channel']}" == from_channel,
+                    )
+                )
             ]
 
         calls = set()
         for row in charm_status:
-            revision, resources = row["Revision"], row["Resources"]
+            revision, resources = int(row["Revision"]), row["Resources"]
             resource_args = (
                 f"--resource={name}:{rev}" for name, rev in resources.items() if rev
             )
             calls.add(
                 (
+                    revision,
                     charm_entity,
                     f"--revision={revision}",
                     *(f"--channel={chan}" for chan in to_channels),
                     *resource_args,
                 )
             )
-        for args in calls:
+
+        # Act on the charm with the highest revision number
+        # This should always be the most recently built charm
+        for _, *args in sorted(calls)[-1:]:
+            # self._echo(" ".join(["charmcraft", "release", *args]))
```

Comment:
this is for local debugging where you don't REALLY want to run the `chamcraft release`